src/OpenTracing/Carriers/HttpHeaders.php

```diff
@@ -26,21 +24,12 @@ public static function fromRequest(RequestInterface $request)
     {
         return new self(
             array_map(function ($values) {
-                return $values[0];
+                return (string) $values[0];
             }, $request->getHeaders())
```

Comment:
This method is guaranteed to return `string[][]` so you shouldn't have to map and cast the values

Reply:
Good point. Changing it.